app/src/main/java/com/example/ithaca_transit_android_v2/NetworkingAdapters.kt

```diff
@@ -44,4 +48,66 @@ class LocationAdapter {
         }
         return DataLocation(final)
     }
+}
+
+
+
+class RouteAdapter{
+    class JsonRoute(
+        val directions: List<Direction>,
+        val startCoords: Coordinate,
+        val endCoords: Coordinate,
+        @Json(name ="arrivalTime")
+        val arrival: Date,
+        @Json(name ="departureTime")
+        val depart: Date
+    )
+
+    @FromJson
+    private fun fromJson(json: JsonRoute): Route {
+        fun computeBoardInMin(firstBusDirection: Direction): Int {
+            val diff = firstBusDirection.startTime.time - Calendar.getInstance().time.time
+            return TimeUnit.MINUTES.convert(diff, TimeUnit.MILLISECONDS).toInt()
+        }
+
+        var firstBus = 0
+        var boardInMins = 0
+        if(json.directions[0].type != DirectionType.BUS){
+            firstBus = 1
+        }
+        if(json.directions.size != 1){
+            boardInMins = computeBoardInMin(json.directions[firstBus])
+        }
```

Comment:
i feel like we can shorten these two values with a ternary or something, ie 
`var firstBus = if json.direction[0].type == DirectionType.BUS 1 else 0`